engine/evalOp.cc

```diff
@@ -831,11 +849,12 @@ namespace minsky
                 if (&to==&from1) checkAllEntriesPresent(to.xVector, from2.xVector);
                 if (&to==&from2) checkAllEntriesPresent(to.xVector, from1.xVector);
             
-                if (from1.numElements()==1 && from2.numElements()==1)
+                // For feature 47            
+                if ((from1.numDenseElements()==1 && from2.numDenseElements()==1) || (from1.numSparseElements==1 && from2.numSparseElements==1))
```

Comment:
You can pretty much guarantee that the single element tensor will be stored densely. Add an assertion to the fact.